tests/functional/test_integration.py

```diff
@@ -1,14 +1,69 @@
 import pytest
+import requests
+import subprocess
+import time
+import yaml
 from ops.model import ActiveStatus
 from pytest_operator.plugin import OpsTest
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test: OpsTest, gunicorn_image, statsd_exporter_image):
+async def test_build_and_deploy_with_psql(
+    ops_test: OpsTest,
+    gunicorn_image,
+    statsd_exporter_image,
+    influx_model_name,
+):
+    result = subprocess.check_output(["juju", "controllers", "--format", "yaml"])
+    controller_name = next(
+        filter(
+            lambda item: item[1]["cloud"] == "localhost",
+            yaml.safe_load(result)["controllers"].items(),
+        )
+    )[0]
+    subprocess.check_output(["juju", "switch", controller_name])
```

Comment:
It should be possible to run juju commands on any controller/ model without needing to switch, e.g., take a look at the `--controller` option here: https://juju.is/docs/olm/juju-add-model

Most Juju commands should have a `--model` and `--controller` option

Lots of examples here: https://github.com/canonical/jenkins-agent-k8s-operator/blob/main/tests/integration/conftest.py